lib/fdoprotctx.c

```diff
@@ -165,14 +163,15 @@ static bool fdo_prot_ctx_connect(fdo_prot_ctx_t *prot_ctx)
 	case FDO_STATE_TO1_SND_PROVE_TO_FDO: /* type 32 */
 		ATTRIBUTE_FALLTHROUGH;
 	case FDO_STATE_TO1_RCV_FDO_REDIRECT: /* type 33 */
-		// try DNS's resolved IP first, if it fails, try given IP address
+		// try DNS's resolved IP first, if it fails, try given IP
+		// address
 		ret = connect_to_rendezvous(
-		    prot_ctx->resolved_ip, prot_ctx->host_dns, prot_ctx->host_port,
-			&prot_ctx->sock_hdl, prot_ctx->tls);
+		    prot_ctx->resolved_ip, prot_ctx->host_dns,
+		    prot_ctx->host_port, &prot_ctx->sock_hdl, prot_ctx->tls);
 		if (!ret) {
 			ret = connect_to_rendezvous(
-				prot_ctx->host_ip, prot_ctx->host_dns, prot_ctx->host_port, 
-				&prot_ctx->sock_hdl, prot_ctx->tls);
+			    prot_ctx->host_ip, NULL, prot_ctx->host_port,
```

Comment:
if resolved_ip is not working, the second call seem to try with host_ip in current implementation. Now we are adding one another fallback combination to the same call. Will it be one another fallback call ?